monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/data/manager/InMemoryDataManager.java

```diff
@@ -0,0 +1,245 @@
+        }
+    }
+
+    @Override
+    public GroupList fetchGroups() {
+        String groupId;
+        try {
+            GroupList groupList = new GroupList();
+            for (HashMap entry: heartBeatStore.values()) {
+                groupId = entry.get("groupId").toString();
+                if (!groupList.contains(groupId)) {
+                    groupList.add(groupId);
+                }
+            }
+            return groupList;
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred fetching groups.", e);
+        }
+    }
+
+    @Override
+    public NodeList fetchNodes(String groupId) {
+        try {
+            NodeList nodeList = new NodeList();
+            for (HashMap entry: serviceInfoStore.values()) {
+                if (entry.get("groupId").toString().equals(groupId)) {
+                    NodeListInner nodeListInner = new NodeListInner();
+                    nodeListInner.setNodeId(entry.get("nodeId").toString());
+                    nodeListInner.setDetails(entry.get("serviceInfo").toString());
+                    nodeList.add(nodeListInner);
+                }
+            }
+            return nodeList;
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred fetching servers.", e);
+        }
+    }
+
+    @Override
+    public String getMgtApiUrl(String groupId, String nodeId) {
+        String mgtApiUrl = "";
+        try  {
+            HashMap valueMap = heartBeatStore.get(groupId + nodeId);
+            mgtApiUrl = valueMap.get("mgtUrl").toString();
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred while retrieveTimestampOfRegisteredNode results.", e);
+        }
+        return mgtApiUrl;
+    }
+
+    @Override
+    public String getAccessToken(String groupId, String nodeId) {
+        String accessToken = "";
+        try {
+            HashMap valueMap = heartBeatStore.get(groupId + nodeId);
+            accessToken = valueMap.get("accessToken").toString();
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred while retrieving access token of node: " + nodeId
+                                               + " in group " + groupId, e);
+        }
+        return accessToken;
+    }
+
+    @Override
+    public String getHeartbeatInterval(String groupId, String nodeId) {
+        try  {
+            HashMap valueMap = heartBeatStore.get(groupId + nodeId);
+            String heartBeatInterval = valueMap.get("interval").toString();
+            return heartBeatInterval;
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred while fetching heartbeat interval of group " + groupId
+                                               + " node " + nodeId, e);
+        }
+    }
+
```

Comment:
Let's remove unnecessary new line here and shall we check other changes as well?

Reply:
fixed